test/metamorpho/TestMetaMorphoSnippets.sol

```diff
@@ -262,6 +268,18 @@ contract TestMetaMorphoSnippets is IntegrationTest {
         assertEq(vault.balanceOf(SUPPLIER), shares, "balanceOf(SUPPLIER)");
     }
 
+    // It covers the branch of _approveMaxVault
+    function testDepositInVault0Approval(uint256 assets) public {
+        assets = bound(assets, MIN_TEST_ASSETS, MAX_TEST_ASSETS);
+        loanToken.setBalance(SUPPLIER, assets);
+        vm.startPrank(SUPPLIER);
+        vault.approve(address(snippets), 0);
+        uint256 shares = snippets.depositInVault(address(vault), assets, SUPPLIER);
```

Comment:
shouln't it revert?

Reply:
no, it forces to use the approveMax written on the snippet. Renamed it into 'testDepositInVaultWithoutPreviousApproval' for clarity, but it is VERY heavy imo...

Reply:
I don't understand what is the purpose of this test. the `SUPPLIER` vault allowance is never used here (the snippets contract doesn't transfer any vault shares from the `SUPPLIER`).

And why should the shares minted be zero ?

Reply:
You were right, fixed in next PR